lua/terrortown/entities/roles/jester/winstates.lua

```diff
@@ -91,57 +91,53 @@ function JesterRevive(ply, fn)
 	end) -- revive after 3s
 end
 
---Player spawns within three seconds with a random opposite role of the killer
+--Player respawns within three seconds with a role in an opposing team of the killer
 function JesterWinstateOne(ply, killer)
 	local rd = killer:GetSubRoleData()
-	local avoidedRoles = {}
+	local reviveRoles = GetSelectableRoles()
 
 	for _, v in pairs(roles.GetList()) do
-		if v.defaultTeam == rd.defaultTeam then
-			avoidedRoles[v] = true
+		if v.defaultTeam == rd.defaultTeam or v.defaultTeam == TEAM_JESTER then
+			reviveRoles[v] = nil
 		end
 	end
 
-	avoidedRoles[ROLE_JESTER] = true
-
 	JesterRevive(ply, function(p)
-		p:SelectRandomRole(avoidedRoles)
+		p:SetRole(reviveRoles[math.random(1, #reviveRoles)])
 		p:SetDefaultCredits()
 
 		SendFullStateUpdate()
 	end)
 end
 
---Player spawns after killer death with a random opposite role
+--Player respawns after killer death with a role in an opposing team of the killer
 function JesterWinstateTwo(ply, killer)
 	local defaultTeam = killer:GetSubRoleData().defaultTeam
 
 	hook.Add("PostPlayerDeath", "JesterWaitForKillerDeath_" .. ply:Nick(), function(deadply)
 		if deadply ~= killer or deadply.NOWINASC then return end
 
-		local avoidedRoles = {}
+		local reviveRoles = GetSelectableRoles()
 
 		for _, v in pairs(roles.GetList()) do
-			if v.defaultTeam == defaultTeam then
-				avoidedRoles[v] = true
+			if v.defaultTeam == defaultTeam or v.defaultTeam == TEAM_JESTER then
+				reviveRoles[v] = nil
```

Comment:
setting an entity of am enumerated table to nil will break the # operator or enumeration. Therefore, you should use table.remove(index).

Reply:
Thanks for letting me know. Although i really don't like using table.remove() performance-wise, i have changed it for now.